script.js

```diff
@@ -506,12 +506,38 @@ close4.addEventListener('click', () => {
 const form = document.getElementById('form');
 const email = document.getElementById('email');
 const error = document.getElementById('error');
+const dame = document.getElementById('same');
+const message = document.getElementById('message');
 
 form.addEventListener('submit', (e) => {
   if (email.value === email.value.toLowerCase()) {
     form.submit();
+    const InputName = dame.value.trim();
+    const InputEmail = email.value.trim();
+    const InputMessage = message.value.trim();
+
+    // set data object
+
+    const myFormData = {
+      name: InputName,
+      email: InputEmail,
+      comment: InputMessage,
+    };
+
+    if (InputName != null || InputEmail != null || InputMessage != null) {
+      return;
+    }
+
+    localStorage.setItem('myFormData', JSON.stringify(myFormData));
```

Comment:
- [ ] It appears that you applied the rule in the wrong place. Indeed it would be best to have the input data persisted in the localStorage after the user has entered those data, not when submitted. The purpose of that behaviour is to allow the entered data not to disappear after a page refresh